src/components/PostDetails.tsx

```diff
@@ -4,31 +4,38 @@ import { NewCommentForm } from './NewCommentForm';
 
 import * as commentsApi from '../api/comments';
 
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import {
+  setComments,
+  addNewComment,
+  removeComment,
+  setLoaded,
+  setHasError,
+} from '../features/commentsSlice';
+
+export const PostDetails: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { selectedPost: post } = useAppSelector(state => state.selectedPost);
+  const {
+    comments,
+    loaded,
+    hasError,
+  } = useAppSelector(state => state.comments);
   const [visible, setVisible] = useState(false);
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
+    dispatch(setLoaded(false));
+    dispatch(setHasError(false));
     setVisible(false);
 
-    commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
+    commentsApi.getPostComments(post?.id || 0)
```

Comment:
Do you need to call API if no post.id? Better check it before and don't call API.